### PR TITLE
Test Quarkus and Camel Infinispan client configuration

### PR DESCRIPTION
**Pull request by jamesnetherton**
Reworking of https://github.com/apache/camel-quarkus/pull/2122.

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: integration-tests/infinispan/src/main/java/org/apache/camel/quarkus/component/infinispan/InfinispanResources.java [hunk 1]
```diff
@@ -35,25 +44,50 @@
...
+
     @PostConstruct
     public void setUp() {
         cacheManager.administration().getOrCreateCache(CACHE_NAME, (String) null);
     }
 
+    @Path("/inspect")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public JsonObject inspect() {
```
Comment (aldettinger) [inline integration-tests/infinispan/src/main/java/org/apache/camel/quarkus/component/infinispan/InfinispanResources.java:62]: I'm not sure about the intent. Would a mehod name like inspectCamelInfinispanClientConfiguration() better reflect the intent ?

File: integration-tests/infinispan/src/test/java/org/apache/camel/quarkus/component/infinispan/InfinispanTest.java [hunk 2]
```diff
@@ -20,21 +20,40 @@
...
+                        "hosts", is(notNullValue()),
+                        "cache-manager", is("none"));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = { "infinispan", "infinispan-quarkus" })
+    public void testInfinispan(String componentName) {
         RestAssured.with()
+                .queryParam("component", componentName)
                 .body("Hello Infinispan")
```
Comment (aldettinger) [inline integration-tests/infinispan/src/test/java/org/apache/camel/quarkus/component/infinispan/InfinispanTest.java:47]: Maybe happening componentName to the body would make the test safer ?

**Review by aldettinger — APPROVED** [review]
2 small neat proposals. Beyond, it looks good to me. Well done :+1:

**Comment by jamesnetherton** [comment]
Latest commit incorporates review suggestions.

**Review by ppalaga — APPROVED** [review]

**Pull request merged**